docs/aws/audit/lambdamonitoring/rules/lambda_step_functions_logging_enabled.mdx

```diff
@@ -22,6 +22,246 @@ CBP,GDPR,HIPAA,ISO27001,SEBI
+4. Select the function you want to check for logging.
+5. In the function overview page, scroll down to the "Monitoring and operations tools" section.
+6. Check if "CloudWatch Logs" is enabled. If it is enabled, you will see a link to "View logs in CloudWatch". If it is not enabled, there will be no such link, indicating that logging is not enabled for this function.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the Lambda functions in your account using the following command:
+
+   ```
+   aws lambda list-functions --region your-region-name
+   ```
+
+   Replace 'your-region-name' with the name of your AWS region. This command will return a JSON object with all your Lambda functions.
+
+3. To check if a specific Lambda function has logging enabled, you need to describe the function configuration using the following command:
+
+   ```
+   aws lambda get-function-configuration --function-name your-function-name --region your-region-name
+   ```
+
+   Replace 'your-function-name' with the name of your Lambda function and 'your-region-name' with the name of your AWS region. This command will return a JSON object with the configuration of your Lambda function.
+
+4. In the returned JSON object, look for the 'TracingConfig' field. If the 'Mode' field under 'TracingConfig' is set to 'Active', then logging is enabled for that function. If it's set to 'PassThrough', then logging is not enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your local environment. This will allow you to interact with AWS services including AWS Lambda.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Use the `boto3` client to interact with AWS Lambda and list all the functions.
+
+```python
+lambda_client = session.client('lambda')
+
+response = lambda_client.list_functions()
+```
+
+4. Iterate over the list of functions and check if the `TracingConfig` is set to `Active`. If it's not, then the function does not have logging enabled.
+
+```python
+for function in response['Functions']:
+    if function['TracingConfig']['Mode'] != 'Active':
+        print(f"Logging is not enabled for function: {function['FunctionName']}")
+```
+
+This script will print out the names of all the Lambda functions that do not have logging enabled. This is a misconfiguration as it's a best practice to enable logging for all Lambda functions to track and debug any issues.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>